### PR TITLE
Use slf4j for main classes, and log4j for tests.

### PR DESCRIPTION
**Pull request by wrandall22**
This is a library, it shouldn't dictate the logging implementation.

### CODE REVIEW

**Comment by wrandall22** [comment]
@mattdrees what do you think?

**Comment by mattdrees** [comment]
Sounds good. When this was created, we didn't have any non-log4j projects. Do we have one, now?

Also, the pom doesn't include a test-scoped slf4j-log4j dep; I'm curious how the tests don't break.

**Comment by wrandall22** [comment]
I'm not sure if we have one or not, but as Cortex is OSGi based, and I'm putting it in there, I thought it would be best to not create more possible conflicts.  I can put a test-scoped slf4j-log4j in there.

File: pom.xml [hunk 1]
```diff
@@ -208,6 +208,13 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>log4j-over-slf4j</artifactId>
```
Comment (mattdrees) [inline pom.xml]: this isn't quite what meant, actually. I meant [slf4j-log4j12](slf4j-log4j12).

`log4j-over-slf4j` goes the other way, making log4j calls turn into slf4j calls.

**Comment by mattdrees** [comment]
thanks; looks good to me

**Comment by wrandall22** [comment]
Should I update the version? I don't think it should break anything in existing systems.

**Comment by mattdrees** [comment]
Sure

Matt

**Pull request merged**